tests/test_pytypes.py

```diff
@@ -261,3 +261,13 @@ def test_number_protocol():
 def test_list_slicing():
     li = list(range(100))
     assert li[::2] == m.test_list_slicing(li)
+
+
+def test_memoryview():
+    import array
+    view = m.test_memoryview(b'abc')
+    if sys.version_info[0] == 3:
+        # Python 2.7 array does not implement the new buffer protocol
```

Comment:
This seems to contradict the documentation:

https://docs.python.org/2/c-api/buffer.html

Quoting: "The new buffer API has been backported to Python 2.6, and the memoryview object has been backported to Python 2.7. It is strongly advised to use them rather than the old APIs, unless you are blocked from doing so for compatibility reasons."

Maybe you need a more if condition here?

Reply:
`array.array` does not offer the new buffer interface, and throws a TypeError.

The documentation states:

> Two examples of objects that support the buffer interface are strings and arrays.
> An array can only expose its contents via the old-style buffer interface. This limitation does not apply to Python 3, where memoryview objects can be constructed from arrays, too.

Unless pybind11 supports the old style buffer interface, it is impossible to accept `array.array` in python 2.7.